api/interventions/views.py

```diff
@@ -172,3 +174,37 @@ def update_status(self, request, pk=None):
                     pk), 404)
 
         return Response(data=response, status=response['status'])
+
+
+    def update(self, request, pk=None):
+        """
+        put:
+        user edit an intervention
+        """
+        try:
+            intervention = InterventionsModel.objects.filter(id=pk)[0]
+        except:
+            return JsonResponse({"status": 404,
+                                 "message": "Intervention with id {} not found".format(pk)},
+                                status=404)
+        serializer = InterventionSerializer(intervention,
+                                            context={'request': request})
+
+        if int(request.user.id) != int(serializer.data['createdBy']):
+            return JsonResponse({"status": 403,
+                                 "message": "You cannot edit an intervention you do not own"},
+                                status=403)
+        request.data['createdBy'] = serializer.data['createdBy']
+
+        if intervention.status.lower() == 'draft':
+            serializer = InterventionSerializer(intervention,
+                                                data=request.data,
+                                                context={'request': request})
+            serializer.is_valid(raise_exception=True)
+            serializer.save()
+            return JsonResponse({"status": 200,
+                                 "data": serializer.data},
+                                status=200)
+        return JsonResponse({"status": 401,
```

Comment:
The status code should be 403; the user is already authorised but forbidden to edit the record since s/he does not own it.

Reply:
Noted and corrected. Thank you